### DISCUSSION TITLE
Footer version numbers

### DISCUSSION THREAD

**Discussion by jSML4ThWwBID69YC** [LemmyNet/lemmy-ui#1497] [General] [answered]
How do you set the version numbers in the footer? 

For example,
_UI: v0.17.3 BE: 0.17.3_

When I build Lemmy/Lemmy-UI, the version comes up as: _BE: unknown version_

Is this a compile time option?

**Comment by dessalines** [comment]
It comes from the docker build, and the git tag:

`echo "pub const VERSION: &str = \"$(git describe --tag)\";" > "crates/utils/src/version.rs" \`

**Reply by sheikharbazalam** [reply]
needs to add the "Version number" to the footer for the FE release currently running.
we have five testing instances and when for example, if i am on testing one and try to show version number for this release in footer,i tried to do some ideas like alter the releasebm or in build script and create seperate script file and call while we build the application to call the script file and create json file to store the release number and then call in footer , but i am not confident in that, please help me with this . We using Angular. we running the build.sh in docker.

**Comment by jSML4ThWwBID69YC** [comment]
Thank you! That sets the BE: version, but not the UI version. Is there a second item for the UI?

**Comment by SleeplessOne1917** [comment]
Only one version is displayed if the  FE and BE versions are the same.

**Comment by jSML4ThWwBID69YC** [comment]
I just rebuilt Lemmy and Lemmy-ui at version 0.17.3. The UI string is still showing unknown version. 

Looks like this
_UI: unknown version BE: 0.17.3_

Any ideas.

**Reply by jheidecker** [reply]
Just my two cents, but IMO, this is not a tag formatting issue. The version that gets displayed is statically set to "unknown version" (https://github.com/LemmyNet/lemmy-ui/blob/0.17.4/src/shared/version.ts) UNLESS you build with Docker (https://github.com/LemmyNet/lemmy-ui/blob/0.17.4/Dockerfile), in which case, it is set to the git tag. So the formatting issue may be true, but not when building UI without using docker. It will just be "unknown version" unless you change it explicitly before building, which then removes it entirely from being displayed IF the string is the same.

**Comment by SleeplessOne1917** [comment]
It seems the UI version should have a v at the front and the backend no v. E.g. v0.17.3. It must have been a mistake when making the tag.

**Comment by jSML4ThWwBID69YC** [comment]
> It seems the UI version should have a v at the front and the backend no v. E.g. v0.17.3. It must have been a mistake when making the tag.

Ok. So it'll probably just work on the next release?

**Comment by SleeplessOne1917** [comment]
> So it'll probably just work on the next release?

Provided a similar mistake doesn’t happen again, yes.

**Comment by jSML4ThWwBID69YC** [comment]
> Provided a similar mistake doesn’t happen again, yes.

Thank you. I appreciate your help.

**Comment by verymilan** [comment]
hmm but if i just install with cargo (following the docs), i can't modify the source anymore to set a version...

edit: i am incorrect , still not very handy

**Comment by dessalines** [comment]
@SleeplessOne1917 we should make those a uniform `0.0.0` with no `v`, and then let the UI add the V if necessary.

Lets leave a lot of these smaller issues until after the http rework.

**Comment by markim** [comment]
I'm doing "Install from scratch" and I can't get the front end to show the right version number. I'm on 0.19.9. The backend works fine, but even manually updating /src/shared/version.ts to a value, it will not show up, always "unknown version". Any help/info is appreciated!

**Comment by jSML4ThWwBID69YC** [comment, marked as answer]
@markim 

I believe this will work.
```
# Build lemmy-ui
git clone https://github.com/LemmyNetlemmy-ui.git --recursive lemmy-ui
cd lemmy-ui && git checkout 0.19.9

# Set version number
echo "export const VERSION = '$(git describe --tag)';" > "src/shared/version.ts"
echo "export const BUILD_DATE_ISO8601 = '$(date -u +"%Y-%m-%dT%H:%M:%SZ")';" > "src/shared/build-date.ts"

# Build
pnpm i
pnpm build:prod
```

**Reply by markim** [reply]
Was able to get it working, thank you!
